test/e2e/egress_services.go

```diff
@@ -891,3 +938,70 @@ func reachAllServiceBackendsFromExternalContainer(container, svcIP string, svcPo
 
 	framework.ExpectEqual(len(backends), 0, fmt.Sprintf("did not reach all pods from outside, missed: %v", backends))
 }
+
+// Sets the "dummy" custom routing table on all of the nodes (this heavily relies on the environment to be a kind cluster)
+// We create a new routing table with 2 routes to the external container:
+// 1) The one from the default routing table.
+// 2) A blackhole with a higher priority
```

Comment:
wouldn't it be better to reverse this logic? In other words the route that "works" is the one in the VRF when the network is used? To make sure the functionality actually works...

A better test would be to add another interface (already done with br-ex2 for exgws) to the node docker containers, put that into the vrf. Then spawn another external container that is only connected to that new docker network and run some server there and verify it receives the packets.

Reply:
> wouldn't it be better to reverse this logic? In other words the route that "works" is the one in the VRF when the network is used? To make sure the functionality actually works...

I wanted to keep the changes on the main routing table as minimal as possible + this mimics the behavior of modifying a custom routing table on the fly.
I think behavior-wise keeping it this way vs. reversing verifies the same stuff (that the custom routing table is actually used), do you think it is better/adds value to do it the opposite?

> A better test would be to add another interface (already done with br-ex2 for exgws) to the node docker containers, put that into the vrf. Then spawn another external container that is only connected to that new docker network and run some server there and verify it receives the packets.

I agree that a more extensive test is better (what you mentioned + perhaps leveraging metallb's vrf) but I rather defer it to a later PR, is that fine?